test/e2e_env/multizone/meshservice/connectivity.go

```diff
@@ -0,0 +1,186 @@
+package meshservice
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	. "github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/framework/client"
+	"github.com/kumahq/kuma/test/framework/deployments/democlient"
+	"github.com/kumahq/kuma/test/framework/deployments/testserver"
+	"github.com/kumahq/kuma/test/framework/envs/multizone"
+)
+
+func Connectivity() {
+	namespace := "msconnectivity"
+	meshName := "msconnectivity"
+
+	BeforeAll(func() {
+		Expect(NewClusterSetup().
+			Install(MTLSMeshUniversal(meshName)).
+			Install(MeshTrafficPermissionAllowAllUniversal(meshName)).
+			Install(YamlUniversal(`
+type: HostnameGenerator
+name: kube-msconnectivity
+spec:
+  template: '{{ .DisplayName }}.{{ .Namespace }}.{{ .Zone }}.msconnectivity'
+  selector:
+    meshService:
+      matchLabels:
+        kuma.io/origin: global
+        kuma.io/env: kubernetes
+`)).
+			Install(YamlUniversal(`
+type: HostnameGenerator
+name: uni-msconnectivity
+spec:
+  template: '{{ .DisplayName }}.{{ .Zone }}.msconnectivity'
```

Comment:
same as above feel free to ignore

```suggestion
  template: '{{ .DisplayName }}.{{ .Zone }}.universal.msconnectivity'
```